### PR TITLE
[BUGFIX] Allow `now()` eval parameter to be used by itself

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- https://github.com/great-expectations/great_expectations/issues/3657
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/core/test_evaluation_parameters.py [hunk 1]
```diff
@@ -133,6 +133,8 @@ def test_parser_timing():
 
 def test_math_evaluation_paramaters():
     assert parse_evaluation_parameter("sin(2*PI)") == math.sin(math.pi * 2)
+    assert parse_evaluation_parameter("cos(2*PI)") == math.cos(math.pi * 2)
+    assert parse_evaluation_parameter("tan(2*PI)") == math.tan(math.pi * 2)
```
Comment (cdkini) [inline tests/core/test_evaluation_parameters.py:137]: Just wanted more coverage here. Not part of the ticket but sneaking it in while I'm in this file.

File: great_expectations/core/evaluation_parameters.py [hunk 2]
```diff
@@ -332,7 +335,12 @@ def parse_evaluation_parameter(
     except ParseException as err:
         L = ["Parse Failure", parameter_expression, (str(err), err.line, err.column)]
 
-    if len(L) == 1 and L[0] not in evaluation_parameters:
+    if len(L) == 1 and isinstance(L[0], tuple) and L[0][-1] is False:
```
Comment (cdkini) [inline great_expectations/core/evaluation_parameters.py]: A valid parser result is a single func with zero args

File: tests/core/test_evaluation_parameters.py [hunk 3]
```diff
@@ -306,3 +308,25 @@ def test_evaluation_parameters_for_between_expectations_parse_correctly(
             "result": {"observed_value": 3},
         }
     )
+
+
+def test_now_evaluation_parameter():
+    """
+    now() is unique in the fact that it is the only evaluation param built-in that has zero arity (takes no arguments).
+    The following tests ensure that it is properly parsed and evaluated in a variety of contexts.
+    """
```
Comment (cdkini) [inline tests/core/test_evaluation_parameters.py:317]: Add test to test_definitions dir before getting this merged!

File: tests/test_definitions/column_map_expectations/expect_column_values_to_be_between__evaluation_parameters.json [hunk 4]
```diff
@@ -119,6 +119,35 @@
...
+            "unexpected_index_list": [0, 1, 2, 3, 4, 5, 6, 7, 9],
+            "success": false
+          },
+          "in": {
+            "column": "ts",
+            "min_value": {
+              "$PARAMETER": "now()"
+            }
+          }
+        },
```
Comment (talagluck) [inline tests/test_definitions/column_map_expectations/expect_column_values_to_be_between__evaluation_parameters.json:150]: Nice!

**Review by talagluck — APPROVED** [review]
LGTM - nice work!

**Pull request merged**